src/drive/web/modules/public/useSharingInfos.spec.js

```diff
@@ -0,0 +1,114 @@
+import React from 'react'
+import { renderHook, act } from '@testing-library/react-hooks'
+
+import { createMockClient } from 'cozy-client'
+
+import { useSharingInfos } from './useSharingInfos'
+import AppLike from 'test/components/AppLike'
+
+describe('useSharingInfos', () => {
+  const { location } = window
+
+  const mockClient = createMockClient({})
+  const getOwnPermissionsMock = jest.fn()
+  const getDiscoveryLinkMock = jest.fn()
+  const queryMock = jest.fn()
+  mockClient.collection = () => ({
+    getOwnPermissions: getOwnPermissionsMock,
+    getDiscoveryLink: getDiscoveryLinkMock
+  })
+  mockClient.query = queryMock
+  const setup = () => {
+    const wrapper = ({ children }) => (
+      <AppLike client={mockClient}>{children}</AppLike>
+    )
+
+    return renderHook(() => useSharingInfos(), {
+      wrapper
+    })
+  }
+
+  const mockSharing = { data: {} }
+  const mockedPermissionsWithInstance = {
+    data: {
+      type: 'io.cozy.permissions',
+      id: 'bf716babb70b73b89c870fccce00233f',
+      attributes: {
+        source_id: 'io.cozy.sharings/bf716babb70b73b89c870fccce00233a'
+      }
+    },
+    included: [
+      {
+        type: 'io.cozy.sharings.members',
+        attributes: {
+          status: 'seen',
+          name: 'Bob',
+          email: 'bob@cozy.tools',
+          instance: 'http://b.cozy.tools'
+        }
+      }
+    ]
+  }
+
+  const mockedPermissionsWithoutInstance = {
+    data: {
+      type: 'io.cozy.permissions',
+      id: 'bf716babb70b73b89c870fccce00233f',
+      attributes: {
+        source_id: 'io.cozy.sharings/bf716babb70b73b89c870fccce00233a'
+      }
+    },
+    included: [
+      {
+        type: 'io.cozy.sharings.members',
+        attributes: {
+          status: 'seen',
+          name: 'Bob',
+          email: 'bob@cozy.tools'
+        }
+      }
+    ]
+  }
+  beforeAll(() => {
+    delete window.location
+    window.location = {
+      pathname: '/preview',
+      search: '?sharecode=5FAZbBB4Iy0k'
+    }
+  })
+  afterAll(() => {
+    window.location = location
+  })
+
+  it('returns the right infos when using useSharingInfo', async () => {
+    const discoveryLink = '/link'
+    getDiscoveryLinkMock.mockReturnValue(discoveryLink)
+    getOwnPermissionsMock.mockResolvedValue(mockedPermissionsWithInstance)
```

Comment:
Nit: I personnally prefer to mock at a lower level, on StackClient::fetchJSON for example. Here we are dependent on the implementation.

Reply:
mocking on fetchJSON makes the test dependent to the implementation too because we'll mock the first call to return X, the second to return Y and the third to return Z. Also it makes. it harder to follow / understand since outside method (I mean in cozy-client for instance) can also makes call to fetchJSON without any notice. 

How do you handle that?

Reply:
When I mock fetchJSON I usually do:

```
fetchJSON.mockImplementation((method, route) => {
  if (method === 'GET' && route == '/permissions') { return mockPermissions; }
  else if (method === 'GET && route == 'otherRoute') { return otherStuff }
  else { throw new Error(`${method}:${route} is not implemented in tests`) }
})
```

Reply:
Interesting, thanks!